src/app/shared/components/template/components/data-items/data-items.component.ts

```diff
@@ -143,6 +129,50 @@ export class TmplDataItemsComponent extends TemplateBaseComponent implements OnD
     });
   }
 
+  /**
+   * When calling set_item or set_items actions from within a data_items loop automatically populate data list
+   * and row id references where not already specified. Additionally add support for populating set_item row id reference
+   * from _index of currently rendered items, and set_items list of ids from currently rendered items
+   */
+  private updateActionList(
+    action_list: FlowTypes.TemplateRowAction[],
+    itemId: string,
+    itemDataIDs: string[]
+  ) {
+    return action_list.map((a) => {
+      // set_item - auto-populate list id and row id parameters from current item context (can be overridden)
+      if (a.action_id === "set_item") {
+        const params: IActionSetItemParams = {
+          _list_id: this.dataListName,
+          _id: itemId,
+          ...a.params,
+        };
+        // if _index used lookup and replace _id
+        if (params._index) {
+          params._id = itemDataIDs[params._index];
```

Comment:
This does not allow for `params._index` to be an expression, e.g. `this.item._index + 1`